code/modules/mob/living/silicon/ai/say.dm

```diff
@@ -119,6 +119,78 @@
 	for(var/word in words) //play vox sounds to the rest of our zlevel
 		play_vox_word(word, src.z, null)
 
+// make sure you don't turn 0 into a false positive
+#define BIOSCAN_DELTA(count, delta) count ? max(0, count + rand(-delta, delta)) : 0
+
+#define AI_SCAN_DELAY 15 SECONDS
+
+///Annonce to everyone the number of xeno and marines on ship and ground
+/mob/living/silicon/ai/proc/ai_bioscan(delta = 1)
+
+	to_chat(usr, span_warning("Scanning for hostile lifeforms..."))
+	if(!do_after(usr, AI_SCAN_DELAY , TRUE, usr, BUSY_ICON_GENERIC)) //initial windup time until firing begins
+		return
+
+	#ifndef TESTING
+	if((last_ai_bioscan + COOLDOWN_AI_BIOSCAN) > world.time)
```

Comment:
err is this just a copypaste
why not make a generic proc in that case

Reply:
I did some experimenting, making bioscan entirely generic runs into a number of issues. Instead I adjusted the AI bioscan action to use the round bioscan instead of copypasting a the proc.